ckanext/datapusher/views.py

```diff
@@ -46,6 +46,16 @@ def get(self, id: str, resource_id: str):
             toolkit.g.pkg_dict = pkg_dict
             toolkit.g.resource = resource
 
+            # Determine if the upload button and delete button
+            # should be enabled or disabled
+            resource_format = resource.get('format', '').strip()
+            supported_formats = toolkit.config.get('ckan.datapusher.formats')
+            is_upload_supported = (
+                resource_format in supported_formats or
+                resource_format.lower() in supported_formats
+            )
+            is_delete_supported = resource.get('datastore_active', False)
```

Comment:
this looks like it will prevent deleting all the resources that have been uploaded to the portal

Reply:
delete from datastore button will be disable only when file is not present to datastore. 
In current scenario, if data is  .not present is datastore, and the user clicks the "Delete from DataStore" button, it results in a 500 Internal Server Error.